File: Easy/Minimum Moves to Sort Permutation/minimum-moves-to-sort-permutation.py
class Solution:
    def minMoves(self, arr):
        """code here"""
        n = len(arr)
        pos = [0] * (n+1)
        
        for i,val in enumerate(arr):
            pos[val] = i
        
        # print (pos)
        maxlen = 1
        curlen = 1
        
        for i in range(2 , n+1):
            if pos[i] > pos[i-1]:
                curlen+=1
            else:
                curlen = 1
            
            maxlen = max(maxlen , curlen)
        
        return n-maxlen

File: Easy/Minimum Moves to Sort Permutation/minimum-moves-to-sort-permutation.cpp
class Solution {
  public:
    int minMoves(vector<int>& arr) {
        // code here
        int n = arr.size();
        
        int pos[n+1];
        
        for (int i = 0 ; i < n ; i++){
            pos[arr[i]] = i;
        }
        
        int ml = 1;
        int cl = 1;
        
        for (int i = 2 ; i <= n ; i++){
            if (pos[i] > pos[i-1]) cl++;
            else cl = 1;
            
            ml = max(ml , cl);
        }
        
        return n-ml;
    }
};